tree/30_generic_methods/file_download.cf

```diff
@@ -0,0 +1,74 @@
+#####################################################################################
+# Copyright 2013 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File download
+# @description Download a file if it does not exit, using curl with a fallback on wget
+#
+# @parameter source      URL to download from
+# @parameter destination File destination
+# 
+# @class_prefix file_download
+# @class_parameter destination
+# This bundle will define a class file_download_${destination}_{kept,repaired,not_ok,ok,reached}
+
+bundle agent file_download(source, destination)
+{
+
+  vars:
+
+    curl_utility_present::
+
+      "action_command"            string => "${unix_knowledge.utility[curl]} -s -L -o ${destination} ${source}";
```

Comment:
For now yes, as there is no "standard" utility on Windows to download files like wget or curl. It can always be added later.